File: 02Fabryka/Students/2019/BieniasJan/Factory/FactoryMethod/Factory.cs
using Factory.MilitaryVehicles;

//Abstract Creator
namespace Factory.FactoryMethod
{
    public abstract class Factory
    {
        protected abstract IMilitaryVehicle Create(string name);

        public IMilitaryVehicle Build(string name)
        {
            IMilitaryVehicle vehicle = Create(name);
            if (vehicle == null)
                return null;
            //vehicle.Shoot();
            //vehicle.Reload();
            //vehicle.Repair();
            return vehicle;
        }
    }
}

File: 02Fabryka/Students/2019/Półgęsek Arkadiusz/factory-method/factory.ts
import { Planet, PlanetType } from "../planets";

export abstract class Factory {
    protected abstract create(type: PlanetType): Planet;
    public build(type: PlanetType): Planet {
        let object = this.create(type);
        if (!object) {
            return null;
        }
        return object;
    }
}